src/Process/Runner/InitialTestsRunner.php

```diff
@@ -66,16 +71,35 @@ public function run(
         array $phpExtraOptions,
         bool $skipCoverage
     ): Process {
+        /** @var Process $process */
         $process = $this->processBuilder->createProcess(
             $testFrameworkExtraOptions,
             $phpExtraOptions,
             $skipCoverage
         );
 
+        //Tracking Process Error Exit
+        $expirationData = (object) [
+            'time' => null,
+        ];
+
         $this->eventDispatcher->dispatch(new InitialTestSuiteWasStarted());
 
-        $process->run(function (string $type) use ($process): void {
+        $process->run(function (string $type) use ($process, $expirationData): void {
             if ($process::ERR === $type) {
+                //If already started, do not start again and let parent one run - prevent infinite loop.
+                //->isRunning calls the callback again every time there is any updated output.
+                if ($expirationData->time !== null) {
```

Comment:
After checking, I'm under the impression this could be done by passing the `$timeout` property to `Process::stop()`. It's however hard to tell for me if it's enough since I can't reproduce the issue

Reply:
I took a look at this and it is unfortunately a little different. By default, the timeout is 10 when calling `stop`. The problem is that `stop` immediately sends the SIGTERM signal and then waits for it to shutdown, which makes sense. This in effect causes the issue we are seeing. If the SIGTERM wasn't there, then this would be fine. BTW, the e2e test is able to replicate the issue easily enough.